### PR TITLE
feat: Typescript 5/4

### PR DESCRIPTION
**Pull request by jimmycallin**
- [ ] I have added automatic tests where applicable
- [ ] The PR title is suitable as a release note
- [ ] The PR contains a description of what has been changed
- [ ] The description contains manual test instructions

## Changes

Additional fixes and enhancements from trying out the beta.

- Exposing all interfaces. I did this by exporting everything (and removing default exports). I don't think this should break anything, but look over the operation changes in particular.
- Added typing to the metadata object in the Response interface

### CODE REVIEW

File: source/operation.ts [hunk 1]
```diff
@@ -61,16 +61,7 @@ export interface GetUploadMetadataOperation {
...
-  | CreateOperation
-  | QueryOperation
-  | SearchOperation
-  | UpdateOperation
-  | DeleteOperation
-  | QueryServerInformationOperation
-  | QuerySchemasOperation
-  | GetUploadMetadataOperation
-  | { action: string; [key: string]: any };
+export type Operation = { action: string; [key: string]: any };
```
Comment (WaylonFtrack) [inline source/operation.ts:64]: Even though they all fall back to this generic type, I think it makes some sense to keep it.

Comment (jimmycallin) [reply source/operation.ts:64]: alright, readded! thanks

File: source/operation.ts [hunk 2]
```diff
@@ -81,7 +72,7 @@ export type Operation =
  * @param  {Object} data Entity data to use for creation
  * @return {Object}      API operation
  */
-export function createOperation(type: string, data: any): CreateOperation {
+export function create(type: string, data: any): CreateOperation {
```
Comment (gismya) [inline source/operation.ts:84]: Even though they probably are not ment to be used outside this project, must we not wait with changing function names until it's time to make a major version release?

Comment (jimmycallin) [reply source/operation.ts:84]: they were not exposed publicly from index.ts, so only way to have used them would be if someone imported them directly from `@ftrack/api/dist/operation`, which I don't think is supported behavior.

Comment (gismya) [reply source/operation.ts:84]: I think I'm just overly cautious.

**Review by gismya — APPROVED** [review]

**Pull request merged**